fabric-lifecycle-events-v1/src/main/java/net/fabricmc/fabric/api/event/lifecycle/v1/ServerLifecycleEvents.java

```diff
@@ -0,0 +1,107 @@
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.event.lifecycle.v1;
+
+import net.minecraft.server.MinecraftServer;
+import net.minecraft.server.world.ServerWorld;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+import net.fabricmc.fabric.impl.lifecycle.ServerLifecycleInternals;
+
+public final class ServerLifecycleEvents {
+	private ServerLifecycleEvents() {
+	}
+
+	/**
+	 * Called when a Minecraft server has started and is about to tick for the first time.
+	 *
+	 * <p>At this stage, all worlds are live.
+	 */
+	public static final Event<ServerLifecycleEvents.LifecycleCallback> SERVER_STARTED = EventFactory.createArrayBacked(ServerLifecycleEvents.LifecycleCallback.class, (callbacks) -> (server) -> {
+		for (ServerLifecycleEvents.LifecycleCallback callback : callbacks) {
+			callback.onChangeLifecycle(server);
+		}
+	});
+
+	/**
+	 * Called when a Minecraft server has started shutting down.
+	 * This occurs before the server's network channel is closed and before any players are disconnected.
+	 *
+	 * <p>For example, an integrated server will begin stopping, but it's client may continue to run.
+	 *
+	 * <p>All worlds are still present and can be modified.
+	 */
+	public static final Event<ServerLifecycleEvents.LifecycleCallback> SERVER_STOPPING = EventFactory.createArrayBacked(ServerLifecycleEvents.LifecycleCallback.class, (callbacks) -> (server) -> {
+		for (ServerLifecycleEvents.LifecycleCallback callback : callbacks) {
+			callback.onChangeLifecycle(server);
+		}
+	});
+
+	/**
+	 * Called when a Minecraft server has stopped.
+	 * All worlds have been closed and all (block)entities and players have been unloaded.
+	 *
+	 * <p>For example, an {@link net.fabricmc.api.EnvType#CLIENT integrated server} will begin stopping, but it's client may continue to run.
+	 * Meanwhile for a {@link net.fabricmc.api.EnvType#SERVER dedicated server}, this will be the last event called.
+	 */
+	public static final Event<ServerLifecycleEvents.LifecycleCallback> SERVER_STOPPED = EventFactory.createArrayBacked(ServerLifecycleEvents.LifecycleCallback.class, callbacks -> server -> {
+		for (ServerLifecycleEvents.LifecycleCallback callback : callbacks) {
+			callback.onChangeLifecycle(server);
+		}
+	});
+
+	/**
+	 * Gets the currently running primary server.
+	 *
+	 * <p><b>Use of this method is highly discouraged and not recommended since there is no real restriction on whether the game engine could run multiple servers concurrently.</b>
+	 * One should attempt to obtain the server instance from a {@link ServerWorld server world} or via other means.
+	 *
+	 * <p>The server instance returned SHOULD NOT be cached! Call the method every time you need the server.
+	 *
+	 * @return the currently running server
+	 * @throws IllegalStateException if the server is not available
+	 */
+	@Deprecated
+	public static MinecraftServer getPrimaryServer() {
```

Comment:
After thinking a bit, I am going to drop these methods.

Reply:
Please do.

Reply:
And it's out